test/sim/items/whiteherb.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe("White Herb", function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should activate after Parting Shot drops both stats, but before the switch is resolved', function () {
+		battle = common.createBattle([[
+			{species: 'torracat', moves: ['partingshot']},
+			{species: 'litten', moves: ['sleeptalk']},
+		], [
+			{species: 'wynaut', item: 'whiteherb', moves: ['sleeptalk']},
+		]]);
+		battle.makeChoices();
+		const wynaut = battle.p2.active[0];
+		assert.false.holdsItem(wynaut);
+		assert.statStage(wynaut, 'atk', 0);
+		assert.statStage(wynaut, 'spa', 0);
+	});
+
+	it.skip('should activate after Abilities that boost stats on KOs', function () {
```

Comment:
Didn’t the bug report say that this is what already happens on sim and that in-game, White Herb should go first? Also, this test is duplicated below.

Reply:
No, it should go like this (Soul-Heart as an example):
Magearna uses Fleur Cannon into target
Special Attack drops to -2
Target is KOed
Soul-Heart boosts to -1
White Herb restores to 0

which is what this test does. I'll fix the duplicate though!

Reply:
In that case, try un-skipping this test. I’m pretty sure this is what the sim does already.

Reply:
Nope, it fails, because the sim is currently restoring the -2 Sp. Atk drop, then applying the Ability KO boost instead of waiting.

``1) White Herb
       should activate after Abilities that boost stats on KOs:
     AssertionError [ERR_ASSERTION]: Expected p2a: Wynaut's spa at stage 0, not at 1.``